new-components/targets/git-clone/pkg/git/git.go

```diff
@@ -0,0 +1,207 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/go-errors/errors"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/smithy-security/pkg/env"
+)
+
+const errInvalidConfigurationReasonEmpty = "cannot be empty"
+
+type (
+	// Conf wraps the component configuration.
+	Conf struct {
+		ClonePath string
+		RepoURL   string
+		Reference string
+
+		ConfAuth ConfAuth
+	}
+
+	// ConfAuth contains authentication configuration.
+	ConfAuth struct {
+		Username    string
+		AuthEnabled bool
+		AccessToken string
+	}
+
+	// Repository wraps a git repository to not leak deps.
+	Repository struct {
+		repo *git.Repository
+	}
+
+	// ErrInvalidConfiguration is used in the constructor.
+	ErrInvalidConfiguration struct {
```

Comment:
why create a custom error and not just use an Errorf? I don't see the extra information being used anywhere

Reply:
Not a big deal, I was thinking of leveraging the error using errors.As but didn't end up doing it - will simplify